client/client.go

```diff
@@ -777,8 +777,7 @@ func (r *NotaryRepository) saveMetadata(ignoreSnapshot bool) error {
 	}
 
 	for role, blob := range targetsToSave {
-		parentDir := filepath.Dir(role)
-		os.MkdirAll(parentDir, 0755)
+		// If the parent directory does not exist, the cache.Set will create it
```

Comment:
I wonder if this code comment is suitable for all the other metadata store implementations like `memorystore` and `httpstore`, they may need not to "create" the parent directory.